executor/executor.go

```diff
@@ -5,175 +5,228 @@ package executor
 
 import (
 	"sync"
+	"sync/atomic"
 
 	"github.com/ava-labs/avalanchego/utils/set"
 
 	"github.com/ava-labs/hypersdk/state"
+
+	uatomic "go.uber.org/atomic"
 )
 
-const defaultSetSize = 8
+// baseDependencies must be greater than the maximum number of dependencies
+// any single task could have. This is used to ensure a dependent task
+// does not begin executing a task until all dependencies have been enqueued.
+const baseDependencies = 100_000_000
```

Comment:
nit: maybe `maxDependencies`? since this number is an upper bound that a task can have and we only decrement our count?

Reply:
We could also just make the `items` argument into the executor have a much stronger invariant instead of using this `const`? Seems like that might be less arbitrary.

Reply:
oh do you mean something like #577 (limiting number of state keys that a tx can specify) ? feel like that would be a stronger invariant